internal/core/command/put.go

```diff
@@ -44,5 +50,7 @@ func NewPutCommand(
 		request.Header.Set(clients.CorrelationHeader, correlationID.(string))
 	}
 
+	request.Header.Set(clients.ContentType, r.Header.Get(clients.ContentType))
```

Comment:
This is duplicated in `get.go`. Can we eliminate code duplication here by introducing a function that adds the necessary headers to the request? Furthermore, I think we can do this in a way that allows us to add more request headers easily in the future. For example, one way would be to have a constant slice of request headers we want to propagate, and the function responsible for adding the headers can iterate over the slice adding any header values that are present in that slice. That way the code will be clean and very easy to add functionality to. I hope that makes sense.